### DISCUSSION TITLE
SelectOne[Button|Menu] display confirm dialog if unsaved data and stop ajax request.

### DISCUSSION THREAD

**Discussion by TheMightyE** [primefaces/community#2117] [PrimeFaces]
Is there a way I can use the JS `confirm()` dialog to capture the input and then continue or cancel the SelectOne's ajax request? I tried it this way and I'm able to stop the remoteCommand execution, but the SelectOne still ends up selecting the new option and also executes the p:ajax `update` and `oncomplete`.
```html
<p:remoteCommand
  name="typeSelectRemoteCommand"
  action="#{bean.onTypeSelect}"
  onstart="PF('statusDialog').show()"
/>
<p:selectOneButton
  value="#{bean.selectedType}"
  onchange="handleTypeSelect()"
  unselectable="false"
>
  <p:ajax
    update="sheet"
    oncomplete="updateSheetSettings(SHEET_OPTIONS); PF('statusDialog').hide(); updateSheetData();"
  />
  <f:selectItems value="#{bean.options}" />
</p:selectOneButton>
```
```javascript
function handleTypeSelect() {
  if (!hasDataChanged()) {
    typeSelectRemoteCommand();
    return;
  }
  if (confirm("You have unsaved changes. Continue?") == true) {
    typeSelectRemoteCommand();
  }
}
```

**Comment by melloware** [comment]
I don't think the confirm dialog can be used this way.  What exactly are you trying to do?

**Reply by TheMightyE** [reply]
My SelectOneButton loads new data for the pe:sheet component depending on the selection. If the user has unsaved changes and tries to select a new option from the SelectOneButton, I want to warn them of unsaved changes and ask for confirmation. If they answer no, then I want the SelectOneButton to keep its current selection.
